### PR TITLE
twingate 2024.191 (new cask)

### PR DESCRIPTION
**Pull request by jeddtdc**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

I think I wrote the uninstall script correctly. Is it also possible to also remove the VPN configuration on an uninstall?

Screenshot 2024-07-16 at 3 29 04 PM

### CODE REVIEW

File: Casks/t/twingate.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  sha256 "89d68bdcae121903a5ea11ffc449b9384314df19a2d48c0b8f6f0424026f5d35"
+
+  url "https://binaries.twingate.com/client/macos/#{version}/Twingate.pkg"
+  name "Twingate"
+  desc "Zero trust network access platform"
+  homepage "https://twingate.com/"
+
+  livecheck do
+    url "https://www.twingate.com/changelog/clients"
+    regex(%r{/macos/(\d+\.\d+\.\d+)/Twingate\.pkg}i)
```
Comment (bevanjkay) [inline Casks/t/twingate.rb:12]:
```suggestion
    regex(%r{/macos/v?(\d+(?:\.\d+)+)/Twingate\.pkg}i)

**Comment by bevanjkay** [comment]
It looks like the following would remove the VPN service
`networksetup -deletepppoeservice Twingate`

This would need to be in the `zap` as a `script` - but there's no current precedence to add this.

**Review by jeddtdc — COMMENTED** [review]
Would something like this work to remove the VPN configuration when `--zap` is included?

File: Casks/t/twingate.rb [hunk 2]
```diff
@@ -22,13 +22,18 @@
...
-    "~/Library/Containers/com.twingate.macos.tunnelprovider",
-    "~/Library/Group Containers/6GX8KVTR9H.com.twingate",
-    "~/Library/Group Containers/group.com.twingate",
-    "~/Library/Preferences/com.twingate.macos.plist",
-  ]
+  zap script: {
+        executable: "/bin/bash",
+        args:       ["-c", "networksetup -deletepppoeservice Twingate"],
+        sudo:       true,
+      },
```
Comment (bevanjkay) [inline Casks/t/twingate.rb:29]:
```suggestion
  zap script: {
            executable: "networksetup",
            args:            ["-deletepppoeservice", "Twingate"],
        },

**Review by p-linnane — APPROVED** [review]

**Comment by bevanjkay** [comment]
Thank you @mooogah

**Pull request merged**